bazel/api-gen/extraction/index.ts

```diff
@@ -1,12 +1,29 @@
 import {writeFileSync} from 'fs';
+import path from 'path';
 // @ts-ignore This compiles fine, but Webstorm doesn't like the ESM import in a CJS context.
 import {NgtscProgram, CompilerOptions, createCompilerHost} from '@angular/compiler-cli';
 
 function main() {
-  const [moduleName, entryPointExecRootRelativePath, srcs, outputFilenameExecRootRelativePath] =
-    process.argv.slice(2);
+  const [
+    moduleName,
+    entryPointExecRootRelativePath,
+    srcs,
+    outputFilenameExecRootRelativePath,
+    serializedPathMapWithExecRootRelativePaths,
+  ] = process.argv.slice(2);
 
-  const compilerOptions: CompilerOptions = {};
+  // The path map is a serialized JSON map of import path to index.ts file.
+  // For example, {'@angular/core': 'path/to/some/index.ts'}
+  const pathMap = JSON.parse(serializedPathMapWithExecRootRelativePaths) as Record<string, string>;
```

Comment:
Note: This is probably fine, but there are limits in how long a shell command can be. That's why we use a Bazel params file most of the time. we can switch to that when we need though.